app/main.py

```diff
@@ -1,2 +1,159 @@
+from fractions import Fraction
+from typing import Any, Hashable, Iterable, Optional
+
+
 class Dictionary:
-    pass
+    class __Node:
+        def __init__(
+                self,
+                key: Optional[Hashable] = None,
+                value: Optional[Any] = None,
+                key_hash: Optional[int] = None
+        ) -> None:
+            self.key = key
+            self.value = value
+            self.key_hash = key_hash
+            self.deleted = False
+
+    def __init__(
+            self,
+            initial_data: Optional[Iterable[tuple[Hashable, Any]]] = None,
+            **kwargs
+    ) -> None:
+        self.__capacity = 8
+        self.__size = 0
+        self.__threshold = int(Fraction(2, 3) * self.__capacity)
+        self.__table = [Dictionary.__Node()] * self.__capacity
+
+        if initial_data:
+            self.update(initial_data)
+        if kwargs:
+            self.update(**kwargs)
+
+    def __repr__(self) -> str:
+        items = ", ".join(f"{key}: {value}" for key, value in self.items())
+        return f"{{{items}}}"
+
+    def __setitem__(self, key: Hashable, value: Any) -> None:
+        index = self.__calculate_index(key)
+
+        if self.__table[index].key is None:
+            if self.__size + 1 > self.__threshold:
+                self.__resize()
+                return self.__setitem__(key, value)
+            self.__size += 1
+
+        self.__table[index] = Dictionary.__Node(key, value, hash(key))
+
+    def __getitem__(self, key: Hashable) -> Any:
+        index = self.__calculate_index(key)
+
+        if self.__table[index].key is None:
+            raise KeyError(key)
+
+        return self.__table[index].value
+
+    def __delitem__(self, key: Hashable) -> None:
+        index = self.__calculate_index(key)
+
+        if self.__table[index].key is None:
+            raise KeyError(key)
+
+        self.__table[index] = Dictionary.__Node()
+        self.__table[index].deleted = True
```

Comment:
you don't really need a deleted attr for Node. If can just set that location in hash table to None. Again this will be possible if you change your implementation for hash table to hold None values

Reply:
I need it because:
1) if there is a colision with _x_ index, I assign the _key, value_ to _x+1_ index
2) then if I delete the key with _x_ - I will not be able to access _x+1_
If I simply set the slot to None, I will lose track of the keys that were stored at the next positions when colision occured.